app/code/community/Wirecard/CheckoutPage/Model/Abstract.php

```diff
@@ -279,7 +279,13 @@ public function getOrder() {
 	 * @return string
 	 */
 	protected function getOrderReference() {
-		return sprintf( '%010d', $this->getOrder()->getRealOrderId() );
+		$orderIncrementId = $this->getOrder()->getRealOrderId();
+		if ( ! is_numeric( $orderIncrementId ))
+		{
+		    return $orderIncrementId;
```

Comment:
Can we please use substring with length of 10 (preferably from the end of the string) so that we don't exceed the max length of this parameter.

Reply:
Sure, this can be done. But be aware that in magento the increment id is not limited to a length of 10 but rather could grow up to 50 chars in length (see `eav_entity_store` table).

Also, in the current implementation `sprintf( '%010d', $this->getOrder()->getRealOrderId() );` the length can already exceed 10. That statement will insert 0's until the string is exactly 10 chars in length. If it is longer than 10 chars the string won't be modified, though, resulting in problems already now.